src/riak_kv_vnode.erl

```diff
@@ -2519,8 +2535,9 @@ encode_handoff_item({B, K}, V) ->
         Value  = riak_object:to_binary_version(?CAP_OBJECT_FORMAT, B, K, V),
         encode_binary_object(B, K, Value)
     catch Error:Reason ->
-            ?LOG_WARNING("Handoff encode failed: ~p:~p",
-                          [Error,Reason]),
+            ?LOG_WARNING(
+                "Handoff encode failed: ~0p:~0p", [Error, Reason]),
+            riak_kv_reader:request_read({B, K}),
```

Comment:
would be nice with a little comment on why this read request is needed here. Not obvious from the code.